### DISCUSSION TITLE
Not working on Svelte 5

### DISCUSSION THREAD

**Discussion by andre-brandao** [revolist/revogrid#579] [General]
**Describe the issue**
Svelte 5 internal pkg got depracated

**To Reproduce**
just npm install on a svelte 5 project, disabling ssr did not solve the issue!

![image](https://github.com/user-attachments/assets/cb2cdc8c-83f5-4912-9eb3-fc2f5a952e2e)

**Comment by m2a2x** [comment]
Hello, I will check asap today, thanks for reporting

**Comment by m2a2x** [comment]
@andre-brandao it took me some time but I've made for you a version with svelte 5 support 4.10.11-next: https://www.npmjs.com/package/@revolist/svelte-datagrid/v/4.10.11-next

I've tested on the latest svelte builds and it's working fine. But be aware version 5 is just release candidate and not the latest version it could lead to multiple issues when used.

**Comment by TimFoerster** [comment]
@m2a2x It does not work for me.

This line:
https://github.com/revolist/svelte-datagrid/blob/04c32934e2baa736f89ee3408fd744e4155fdafd/lib/components/RevoGrid.ts#L535
Throws me an error.

It compiles to 
```
  slot(node, (void 0)($$props), {}, null);
```

Resulting in:
`TypeError: (void 0) is not a function.`

svelte file:
```html
<script lang="ts">
	import type { PageData } from './$types';
	import { RevoGrid, type ColumnRegular } from '@revolist/svelte-datagrid';

	let { data }: { data: PageData } = $props();
	import { onMount } from 'svelte';

	const source = [
		{
			name: '1',
			details: 'Item 1'
		},
		{
			name: '2',
			details: 'Item 2'
		}
	];
	const columns: ColumnRegular[] = [
		{
			prop: 'name',
			name: 'First'
		},
		{
			prop: 'details',
			name: 'Second'
		}
	];

	let mounted = $state(false);
	onMount(() => {
		mounted = true;
	});
</script>

{#if mounted}
	<RevoGrid {source} {columns}></RevoGrid>
{/if}
```

with
```
@revolist/svelte-datagrid: "4.10.11-next"
@sveltejs/kit:  "3.3.1"
"svelte": "5.2.7"
```

**Comment by revolist** [comment]
Hi, made an example for svelte kit and improved documentation
https://rv-grid.com/guide/svelte/
https://github.com/revolist/revogrid-svelte-kit
